### PR TITLE
Add Fortran 2008 syntax to the manpages

### PR DESCRIPTION
**Pull request by ggouaillardet**
Refers #959

### CODE REVIEW

**Comment by hjelmn** [comment]
I see at least one missing one: MPI_Win_shared_query. There might be others. Otherwise :+1:.

**Comment by hjelmn** [comment]
Hmm, sorry. That one is in there. Just didn't show up in the github diff.

I did a quick check and the following files that have f08 were not modified:

```
MPI_Aint_add.3in
MPI_Wtick.3in (no difference between use mpi and use mpif08 but might be worth showing the use mpif08)
MPI_Wtime.3in (no difference between use mpi and use mpif08 but might be worth showing the use mpif08)
```

The following were not modified but don't need to be:

```
MPI_Address (removed MPI-3.0)
MPI_Attr_delete (deprecated)
MPI_Attr_get (deprecated)
MPI_Attr_put (deprecated)
MPI_Comm_f2c (no fortran)
MPI_Errhandler_create (removed MPI-3.0)
MPI_Errhandler_get (removed MPI-3.0)
MPI_Errhandler_set (removed MPI-3.0)
MPI_Keyval_create (deprecated)
MPI_Keyval_free (deprecated)
MPI_Status_f2c (no fortran)
MPI_Type_extent (removed MPI-3.0)
MPI_Type_hindexed (removed MPI-3.0)
MPI_Type_hvector (removed MPI-3.0)
MPI_Type_lb (removed MPI-3.0)
MPI_Type_struct (removed MPI-3.0)
MPI_Type_ub (removed MPI-3.0)

```

**Comment by jsquyres** [comment]
@ggouaillardet Can you have a look at @hjelmn's comment?

**Comment by ggouaillardet** [comment]
@jsquyres i just pushed my new changes based on @hjelmn comment, and i added an other commit to trim the man pages (if this is not needed, i am fine with that)
btw, this PR has both blocker and enhancement labels (!) i guess blocker is not needed here

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by jsquyres** [comment]
I put a blocker label on it because I'd really like to see this stuff in 2.0.0.  :smile: 

@hppritcha There was a failure in the distcheck jenkins -- it complains about make_manpage.pl not being found.  Is that a real error?

**Comment by ggouaillardet** [comment]
I likely forgot to add this new script into the distrib,
will fix that tomorrow

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by ggouaillardet** [comment]
:bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

File: Makefile.ompi-rules [hunk 1]
```diff
@@ -8,32 +8,50 @@
...
+	@ $(top_srcdir)/ompi/mpi/man/make_manpage.pl \
+	  --package-name='@PACKAGE_NAME@' \
+	  --package-version='@PACKAGE_VERSION@' \
+	  --ompi-date='@OMPI_RELEASE_DATE@' \
+	  --opal-date='@OPAL_RELEASE_DATE@' \
+	  --orte-date='@ORTE_RELEASE_DATE@' \
+          $(TRIM_OPTIONS) \
+	  --input=$< \
+	  --output=$@
+
```
Comment (jsquyres) [inline Makefile.ompi-rules]: Watch the excess whitespace at the end of the line here.

**Comment by jsquyres** [comment]
One other minor/meta comment: should we change

```
INCLUDE 'mpif.h'
```

to

```
USE MPI
! or the older form: INCLUDE 'mpif.h'
```

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by ggouaillardet** [comment]
:bot:retest

**Comment by ggouaillardet** [comment]
@jsquyres i made the requested change

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by ggouaillardet** [comment]
@hppritcha it looks like a local issue
can you please have a look ?

**Comment by jsquyres** [comment]
:+1:

**Comment by jsquyres** [comment]
@ggouaillardet @hppritcha Looks like a local failure in LANL Cray XC.

**Pull request merged**